gooddata-sdk/gooddata_sdk/catalog/workspace/declarative_model/workspace/workspace.py

```diff
@@ -0,0 +1,217 @@
+# (C) 2022 GoodData Corporation
+from __future__ import annotations
+
+from typing import Any
+
+from gooddata_metadata_client.model.declarative_workspace import DeclarativeWorkspace
+from gooddata_metadata_client.model.declarative_workspace_data_filter import DeclarativeWorkspaceDataFilter
+from gooddata_metadata_client.model.declarative_workspace_data_filter_setting import (
+    DeclarativeWorkspaceDataFilterSetting,
+)
+from gooddata_metadata_client.model.declarative_workspace_model import DeclarativeWorkspaceModel
+from gooddata_metadata_client.model.declarative_workspaces import DeclarativeWorkspaces
+from gooddata_sdk.catalog.entity import CatalogNameEntity, CatalogTitleEntity
+from gooddata_sdk.catalog.identifier import CatalogWorkspaceIdentifier
+from gooddata_sdk.catalog.workspace.declarative_model.workspace.analytics_model.analytics_model import (
+    CatalogDeclarativeAnalyticsLayer,
+)
+from gooddata_sdk.catalog.workspace.declarative_model.workspace.logical_model.ldm import CatalogDeclarativeLdm
+
+
+class CatalogDeclarativeWorkspaceModel:
+    def __init__(self, ldm: CatalogDeclarativeLdm = None, analytics: CatalogDeclarativeAnalyticsLayer = None):
+        self.ldm = ldm
+        self.analytics = analytics
+
+    @classmethod
+    def from_api(cls, entity: dict[str, Any]) -> CatalogDeclarativeWorkspaceModel:
+        ldm, analytics = None, None
+        if entity.get("ldm"):
```

Comment:
In reality, this can't happen.
Even when workspace is completely empty, backend returns ldm/analytics keys with subkeys down to first subkey containing array, which is empty.
But keep it like it is, so we are safe even in case of breaking changes in backend.